CMakeLists.txt

```diff
@@ -63,11 +63,18 @@ if(NOT DEFINED CMAKE_BUILD_TYPE)
 endif()
 
 project(JSONCPP
-        VERSION 1.9.0 # <major>[.<minor>[.<patch>[.<tweak>]]]
+        # Note: version must be updated in three places when doing a release. This
+        # annoying process ensures that amalgamate, CMake, and meson all report the
+        # correct version.
+        # 1. /meson.build
+        # 2. /include/json/version.h
+        # 3. /CMakeLists.txt
+        # IMPORTANT: also update the SOVERSION!!
+        VERSION 1.9.2 # <major>[.<minor>[.<patch>[.<tweak>]]]
```

Comment:
CMake's file command can read a variable from a file.

    file (STRINGS "version.txt" JSONCPP_VERSION)
    file (STRINGS "soversion.txt" JSONCPP_SOVERSION)

I don't know how to do it in meson.
I guess a worst-case would be invoking cat.

    r = run_command('cat','version.txt')
    JSONCPP_VERSION=r.stdout()

    r = run_command('cat','soversion.txt')
    JSONCPP_SOVERSION=r.stdout()

I don't know meson. It feels like we should be able to do this though!

Reply:
I'll spend some time this week looking into that, I definitely agree with you.